lazysnap/src/main/java/com/google/accompanist/lazysnap/SnappingFlingBehavior.kt

```diff
@@ -0,0 +1,689 @@
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.truncate
+
+private const val DebugLog = false
+
+@RequiresOptIn(message = "Accompanist Lazy Snap is experimental. The API may be changed in the future.")
+@Retention(AnnotationRetention.BINARY)
+annotation class ExperimentalLazySnapApi
+
+/**
+ * Default values used for [SnappingFlingBehavior] & [rememberSnappingFlingBehavior].
+ */
+@ExperimentalLazySnapApi
+object SnappingFlingBehaviorDefaults {
+    /**
+     * [AnimationSpec] used as the default value for the `snapAnimationSpec` parameter on
+     * [rememberSnappingFlingBehavior] and [SnappingFlingBehavior].
+     */
+    val snapAnimationSpec: AnimationSpec<Float> = spring(stiffness = 400f)
+
+    /**
+     * The default implementation for the `maximumFlingDistance` parameter of
+     * [rememberSnappingFlingBehavior] and [SnappingFlingBehavior], which does not limit
+     * the fling distance.
+     */
+    val maximumFlingDistance: (LazyListLayoutInfo) -> Int = { Int.MAX_VALUE }
+}
+
+/**
+ * Create and remember a snapping [FlingBehavior] to be used with [LazyListState].
+ *
+ * @param lazyListState The [LazyListState] to update.
+ * @param snapOffsetForItem Block which returns which offset the given item should 'snap' to.
+ * See [SnapOffsets] for provided values.
+ * @param maximumFlingDistance Block which returns the maximum fling distance in pixels.
+ * The returned value should be >= 0.
+ * @param decayAnimationSpec The decay animation spec to use for decayed flings.
+ * @param snapAnimationSpec The animation spec to use when snapping.
+ * @param endContentPadding The amount of content padding on the end edge of the lazy list
+ * in pixels (end/bottom depending on the scrolling direction).
+ */
+@ExperimentalLazySnapApi
+@Composable
+fun rememberSnappingFlingBehavior(
+    lazyListState: LazyListState,
+    snapOffsetForItem: (layoutInfo: LazyListLayoutInfo, itemInfo: LazyListItemInfo) -> Int = SnapOffsets.Center,
+    maximumFlingDistance: (LazyListLayoutInfo) -> Int = SnappingFlingBehaviorDefaults.maximumFlingDistance,
+    decayAnimationSpec: DecayAnimationSpec<Float> = rememberSplineBasedDecay(),
+    snapAnimationSpec: AnimationSpec<Float> = SnappingFlingBehaviorDefaults.snapAnimationSpec,
+    @Px endContentPadding: Int = 0,
+): SnappingFlingBehavior = remember(
+    lazyListState,
+    snapOffsetForItem,
+    maximumFlingDistance,
+    decayAnimationSpec,
+    snapAnimationSpec,
+) {
+    SnappingFlingBehavior(
+        lazyListState = lazyListState,
+        snapOffsetForItem = snapOffsetForItem,
+        maximumFlingDistance = maximumFlingDistance,
+        decayAnimationSpec = decayAnimationSpec,
+        snapAnimationSpec = snapAnimationSpec,
+    )
+}.apply {
+    this.endContentPadding = endContentPadding
+}
+
+/**
+ * Contains a number of values which can be used for the `snapOffsetForItem` parameter on
+ * [rememberSnappingFlingBehavior] and [SnappingFlingBehavior].
+ */
+@Suppress("unused")
```

Comment:
why is it needed?

Reply:
Because library mode doesn't work in Android projects, so we can't declare something as public API. Some of vals in the class aren't used in the project, so lint complains that we can remove them. Added a comment explaining.